tests/unit/hykkt/HykktPermutationTests.hpp

```diff
@@ -1,130 +1,280 @@
+
+        pc.addHInfo(hes);
+        pc.addJInfo(jac);
+        pc.addJtInfo(jac_tr);
+
         pc.addPerm(perm);
         pc.invertPerm();
 
         // Test RC permutation
-        pc.vecMapRC(b_i, b_j);
-        printf("Comparing RC permutation\n");
-        for (int i = 0; i < n + 1; i++) // Loop over row pointers (n+1)
+        pc.vecMapRC(result_prc_i, result_prc_j);
+        printf("Comparing RC permutation of H\n");
+        bool flagrc = verifyResults(hes_prc_i, result_prc_i, n + 1);
+        flagrc &= verifyResults(hes_prc_j, result_prc_j, nnz_hes);
+        printf(!flagrc ? "RC permutation failed\n" : "RC permutation passed\n");
+
+        // Test R permutation
+        pc.vecMapR(result_pr_i, result_pr_j);
+        printf("Comparing R permutation of J_tr\n");
+        bool flagr = verifyResults(jac_tr_pr_i, result_pr_i, m + 1);
+        flagr &= verifyResults(jac_tr_pr_j, result_pr_j, nnz_jac);
+        printf(!flagr ? "R permutation failed\n" : "R permutation passed\n");
+
+        // Test C permutation
+        pc.vecMapC(result_pc_j);
+        printf("Comparing C permutation of J\n");
+        bool flagc = verifyResults(jac_pc_j, result_pc_j, nnz_jac);
+        printf(!flagc ? "C permutation failed\n" : "C permutation passed\n");
+
+        double hes_prc_v[6]   = {4, 5, 1, 0, 3, 2};
+        double jac_pc_v[4]    = {1, 0, 3, 2};
+        double jac_tr_pr_v[4] = {2, 3, 0, 1};
+
+        // Test mapIndex on H
+        double* result_prc_v = allocateArray(nnz_hes);
+        pc.mapIndex(ReSolve::hykkt::PERM_HES_V, hes->getValues(memspace_), result_prc_v);
+        double* h_result_prc_v = bringToHost(result_prc_v, nnz_hes);
+        printf("Comparing mapped H nonzero values\n");
+        bool flagrc_v = verifyResults(hes_prc_v, h_result_prc_v, nnz_hes);
+        printf(!flagrc_v ? "Map Index failed on H\n" : "Map Index passed on H\n");
+
+        // Test mapIndex on J
+        double* result_pc_v = allocateArray(nnz_jac);
+        pc.mapIndex(ReSolve::hykkt::PERM_JAC_V, jac->getValues(memspace_), result_pc_v);
+        double* h_result_pc_v = bringToHost(result_pc_v, nnz_jac);
+        printf("Comparing mapped J nonzero values\n");
+        bool flagc_v = verifyResults(jac_pc_v, h_result_pc_v, nnz_jac);
+        printf(!flagc_v ? "Map Index failed on J\n" : "Map Index passed on J\n");
+
+        // Test mapIndex on J transpose
+        double* result_pr_v = allocateArray(nnz_jac);
+        pc.mapIndex(ReSolve::hykkt::PERM_JAC_TR_V, jac_tr->getValues(memspace_), result_pr_v);
+        double* h_result_pr_v = bringToHost(result_pr_v, nnz_jac);
+        printf("Comparing mapped J_TR nonzero values\n");
+        bool flagr_v = verifyResults(jac_tr_pr_v, result_pr_v, nnz_jac);
+        printf(!flagr_v ? "Map Index failed on J_TR\n" : "Map Index passed on J_TR\n");
+
+        double  h_indices[3] = {0, 1, 2};
+        double* indices      = allocateArray(n);
```

Comment:
```suggestion
        double* indices;
```
this doesn't appear to be necessary, we overwrite it immediately after this, resulting in a memory leak.